### PR TITLE
Added #1891: Set ping packet size

### LINKED ISSUE #1891: Ping packet size
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Notification

### 🔖 Feature description

If devices only repond to pings from windows machines:

Change ping packet size with -s option.
Some devices I had only reply with ping size of 32 not 56 which linux uses.

### ✔️ Solution

(I did this for my installation :)  

 } else if (util.LIN) {
        this._bin = "/bin/ping";

        const defaultArgs = [ "-n", "-s", "32", "-w", timeout, "-c", "1", host ];

        if (net.isIPv6(host) || options.ipv6) {
            defaultArgs.unshift("-6");
        }

        this._args = (options.args) ? options.args : defaultArgs;
        this._regmatch = /=([0-9.]+?) ms/;

### ❓ Alternatives

modding my own install

### 📝 Additional Context

rare

### PR DESCRIPTION
**Pull request by Computroniks**
# Description
This should fully implement https://github.com/louislam/uptime-kuma/issues/1891 by adding an extra field to the edit
monitor page and an extra column to the database. The user can now
set the size of the packet to send, it defaults to 56. A maximum limit
of 65500 was chosen to ensure that the total size of the packet does
not exceed the IPv4 maximum packet size and to comply with the limit
imposed by Windows.

Fixes #1891 

## Type of change

Please delete any options that are not relevant.

- User interface (UI)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings

## Screenshots
image
Wireshark capture showing different packet sizes

image
Changes to UI

### CODE REVIEW

**Comment by chakflying** [comment]
The default on the UI side is set to 32.

**Comment by Computroniks** [comment]
> The default on the UI side is set to 32.

Oops. Thats was a leftover from whem I initially set the default to 32 to match windows instead of 56 to match Linux. Most recent commit should have fixed that.

File: server/util-server.js [hunk 1]
```diff
@@ -89,12 +90,14 @@ exports.ping = async (hostname) => {
...
+ * @param {number} [size=56] Size of ping packet to send
  * @returns {Promise<number>} Time for ping in ms rounded to nearest integer
  */
-exports.pingAsync = function (hostname, ipv6 = false) {
+exports.pingAsync = function (hostname, ipv6 = false, size = 56) {
     return new Promise((resolve, reject) => {
         const ping = new Ping(hostname, {
-            ipv6
+            ipv6,
+            size
```
Comment (Saibamen) [inline server/util-server.js:100]:
```suggestion
            size,
```

**Comment by louislam** [comment]
Might need help to test it on MacOS and FreeBSD.

**Comment by Computroniks** [comment]
Fixed merge conflicts

**Comment by louislam** [comment]
> Might need help to test it on MacOS and FreeBSD.

Anyone could help with this?

**Comment by iagosrodrigues** [comment]
> > Might need help to test it on macOS and FreeBSD.
> 
> Anyone could help with this?

Ping from macOS is 56 bytes, like Linux.

**Comment by Computroniks** [comment]
Fixed merge conflicts. Re #2223  - I think this should be merged first and then the change in module merged after, simply because I won't be able to work on that for a week or two and it would probably make it easier for me when switching the module.

**Comment by Computroniks** [comment]
Converted to draft to update to match #2223

**Comment by Computroniks** [comment]
This has now been updated to work with the changes made in #2223 and should be ready for review. I tested the changes and they work on Linux.

**Pull request merged**